src/Libraries/DesignScriptBuiltin/Dictionary.cs

```diff
@@ -13,9 +13,15 @@ public class Dictionary
         {
             private readonly ImmutableDictionary<string, object> D;
 
+            /// <summary>
+            /// Use backingDict to read from D.
```

Comment:
I was thinking the same thing. Are you suggesting we return new dictionaries (like we return new geometry in LibG) from modifier methods such as `SetValueAtKeys`, `RemoveKeys`, etc.?

Reply:
yes, I guess we'd have to, and I imagine we'd need to add some tests to guarantee that new dictionaries are always returned so that contract previously afforded by immutable.dictionary is not broken in the future.

Reply:
So you're still in favor of doing that over keeping immutable dictionaries and maintaining a separate dictionary? Is that mainly because of having one less dependency on system.collections.immutable or something more?

Reply:
Less dependencies, and it's just not obvious what benefit this type gives us beside the internal contract. Is it faster than just using Dictionaries and copying them?